### PR TITLE
Unify priorities for connection pools

### PR DESCRIPTION
**Pull request by serxa**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 2df0e0c66962ce91e47eb4c98bae4fabe1ce9cc1 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50675/2df0e0c66962ce91e47eb4c98bae4fabe1ce9cc1/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Client/ConnectionPool.cpp [hunk 1]
```diff
@@ -18,7 +18,7 @@ ConnectionPoolPtr ConnectionPoolFactory::get(
     String client_name,
     Protocol::Compression compression,
     Protocol::Secure secure,
-    Int64 priority)
+    Priority priority)
```
Comment (kssenii) [inline src/Client/ConnectionPool.cpp:21]: but does the comment from `Common/Priority.h` apply in all these places "lower value means higher priority"? I am not sure about that, should we remove the comment then?

Comment (serxa) [reply src/Client/ConnectionPool.cpp:21]: Yes, the idea is to use the same logic everywhere, am I wrong this is the case for priority here?

Comment (kssenii) [reply src/Client/ConnectionPool.cpp:21]: Indeed looks so, just asked because I did not know if all of them follow the same priority rule

**Review by kssenii — APPROVED** [review]

**Pull request merged**